vng_api_common/generators.py

```diff
@@ -29,110 +20,18 @@ def get_allowed_methods(self, callback) -> list:
-        )
-        self.consumes = get_consumes(api_settings.DEFAULT_PARSER_CLASSES)
-        self.produces = get_produces(api_settings.DEFAULT_RENDERER_CLASSES)
-        paths, prefix = self.get_paths(endpoints, components, request, public)
-
-        security_definitions = self.get_security_definitions()
-        if security_definitions:
-            security_requirements = self.get_security_requirements(security_definitions)
-        else:
-            security_requirements = None
-
-        url = self.url
-        if url is None and request is not None:
-            url = request.build_absolute_uri()
-
-        return openapi.Swagger(
-            info=self.info,
-            paths=paths,
-            consumes=self.consumes or None,
-            produces=self.produces or None,
-            tags=self.get_tags(request, public),
-            security_definitions=security_definitions,
-            security=security_requirements,
-            _url=url,
-            _prefix=prefix,
-            _version=self.version,
-            **dict(components),
-        )
-
-    def get_path_parameters(self, path, view_cls):
-        """Return a list of Parameter instances corresponding to any templated path variables.
-
-        :param str path: templated request path
-        :param type view_cls: the view class associated with the path
-        :return: path parameters
-        :rtype: list[openapi.Parameter]
+        workaround for HEAD method which doesn't have action
         """
-        parameters = super().get_path_parameters(path, view_cls)
-
-        # see if we can specify UUID a bit more
-        for parameter in parameters:
-            # the most pragmatic of checks
-            if not parameter.name.endswith("_uuid"):
-                continue
-            parameter.format = openapi.FORMAT_UUID
-            parameter.description = "Unieke resource identifier (UUID4)"
-        return parameters
-
-    def get_operation_keys(self, subpath, method, view) -> List[str]:
-        if method != "HEAD":
-            return super().get_operation_keys(subpath, method, view)
-
-        assert not is_list_view(
-            subpath, method, view
-        ), "HEAD requests are only supported on detail endpoints"
-
-        # taken from DRF schema generation
-        named_path_components = [
-            component
-            for component in subpath.strip("/").split("/")
-            if "{" not in component
-        ]
+        if method == "HEAD":
+            view = super(_OpenAPISchemaGenerator, self).create_view(
+                callback, method, request=request
+            )
+            return view
 
-        return named_path_components + ["headers"]
+        return super().create_view(callback, method, request=request)
 
-    def get_overrides(self, view, method) -> dict:
-        if method == "HEAD":
-            return {}
-        return super().get_overrides(view, method)
+    # todo support registering and reusing Response components
```

Comment:
Is this meant to be picked up by a separate ticket?

Reply:
I've created an issue for it - https://github.com/maykinmedia/commonground-api-common/issues/55